.github/workflows/pr-verify.yml

```diff
@@ -0,0 +1,178 @@
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check if UI files changed
+        id: ui-changes
+        run: |
+          git fetch origin main
+          if git diff --name-only origin/main...HEAD | grep '^ui/'; then
+            echo "changes=true" >> $GITHUB_ENV
+          else
+            echo "changes=false" >> $GITHUB_ENV
+          fi
+
+      - name: Setup Node ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v4
+        if: env.changes == 'true'
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: 'npm'
+          cache-dependency-path: ui/package-lock.json
+
+      - name: Download results from generate-resources job
+        uses: actions/download-artifact@v4
+        with:
+          name: ui-autogenerated
+          path: ui/autogenerated
+
+      - name: Install dependencies
+        if: env.changes == 'true'
+        working-directory: ui
+        run: npm ci
+
+      - name: Build UI
+        if: env.changes == 'true'
+        working-directory: ui
+        run: npm run build
+  
+  verify-backend:
+    name: Verify backend
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check if backend files changed
+        id: backend-changes
+        run: |
+          git fetch origin main
+          if git diff --name-only origin/main...HEAD | grep '^backend/'; then
+            echo "changes=true" >> $GITHUB_ENV
+          else
+            echo "changes=false" >> $GITHUB_ENV
+          fi
+
+      - name: Setup Node ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v4
+        if: env.changes == 'true'
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: 'npm'
+          cache-dependency-path: backend/package-lock.json
+
+      - name: Install dependencies
+        working-directory: backend
+        if: env.changes == 'true'
+        run: npm ci
+
+      - name: Lint backend
+        working-directory: backend
+        if: env.changes == 'true'
+        run: npm run lint
+
+  verify-docker:
+    name: Verify docker build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check if docker files changed
+        id: dockerfiles-changes
+        run: |
+          git fetch origin main
+          if git diff --name-only origin/main...HEAD | grep -E '^(Dockerfile|docker-compose\.yaml|\.dockerignore)$'; then
+            echo "changes=true" >> $GITHUB_ENV
+          else
+            echo "changes=false" >> $GITHUB_ENV
+          fi
+
+      - name: Login to Application Collection
+        uses: docker/login-action@v3
+        if: env.changes == 'true'
+        with:
+          registry: dp.apps.rancher.io
+          username: ${{ secrets.APPLICATION_COLLECTION_USERNAME }}
```

Comment:
```suggestion
          username: ${{ secrets.APPCO_SA_USERNAME }}
```

Reply:
We are already using this key in other CI/CD pipelines.

Reply:
Done 6f145d1